src/lazy/never.rs

```diff
@@ -78,87 +78,3 @@ impl MakeValueWriter for Never {
 }
 
 impl MacroArgsWriter for Never {}
-
-impl ValueWriter for Never {
```

Comment:
🗺️ `Never` implements many traits so it can be used as a marker to statically indicate unreachable/unused code paths. Previously, `Never` implemented both `ValueWriter` and `AnnotatableValueWriter`; the new blanket impl allows us to eliminate the explicit implementation of `ValueWriter`.